lib/experimental/fonts-api/class-wp-fonts-resolver.php

```diff
@@ -202,6 +202,11 @@ private static function get_settings() {
 				$settings            = static::set_tyopgraphy_settings_array_structure( $settings );
 			}
 
+			// Skip if typography.fontFamilies.theme from the global settings and the variation are not arrays.
+			if ( ! is_array( $settings['typography']['fontFamilies']['theme'] ) && ! is_array( $variation['settings']['typography']['fontFamilies']['theme'] ) ) {
```

Comment:
Seems this should be `||` not `&&` as one of these may be null or perhaps not set? Also perhaps checking them separately may be better? Then the `$settings['typography']['fontFamilies']['theme']` can probably be above the loop as it doesn't change.

Reply:
This makes sense, thank you!

I noticed https://github.com/WordPress/gutenberg/pull/55981 is probably closer to a better fix, so happy to close this PR is favour of that. Or also happy to continue working on this one if it ends up being needed.